gobblin-data-management/src/main/java/gobblin/data/management/retention/policy/HiveTablePartitionTimeBasedRetentionPolicy.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.data.management.retention.policy;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Properties;
+
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.ql.metadata.Partition;
+import org.apache.hadoop.hive.ql.metadata.Table;
+import org.joda.time.DateTime;
+import org.joda.time.Duration;
+
+import gobblin.data.management.version.DatasetVersion;
+import gobblin.data.management.version.HiveTablePartitionVersion;
+import gobblin.util.Either;
+
+
+/**
+ * List deletable {@link HiveTablePartitionVersion}s by checking if the modification timestamp of the data location older than currentTime minus {@link #retention}.
+ * This can be overridden by getting the time information from other ways, e.g., the datetime pattern along the path.
+ */
+public class HiveTablePartitionTimeBasedRetentionPolicy implements RetentionPolicy<HiveTablePartitionVersion> {
```

Comment:
I think we also need a retention policy that uses information in the partition values itself. For example, if the partition value is `datepartition="2016-01-01"`, we should be able to run retention on that date instead of modtime.